pipeline/api/cloud.py

```diff
@@ -169,37 +198,20 @@ def _get_raise_for_status(response: httpx.Response) -> None:
             else:
                 response.raise_for_status()
 
-    def upload_file(self, file_or_path) -> FileGet:
-
-        if isinstance(file_or_path, str):
-            # TODO: Change this to wrap the file object reader to convert to hex
-            # everytime anything is read instead of reading it all at once.
-
-            with open(file_or_path, "rb") as file:
-                buffer = file.read()
-            hex_buffer = buffer.hex()
-            return self._post_file(
-                "/v2/files/",
-                io.BytesIO(hex_buffer.encode()),
-            )
-        else:
-            return self._post_file("/v2/files/", file_or_path)
-
-    def upload_data(self, file_or_path) -> DataGet:
-        uploaded_file = self.upload_file(file_or_path)
-        uploaded_data = self._post("/v2/data", uploaded_file.dict())
+    def upload_data(self, obj: Any) -> DataGet:
+        """Upload the object as a pickle-encoded file to the API."""
+        files = dict(data=_as_upload_file(_as_binary_file(obj)))
```

Comment:
it seems like we always call `_as_upload_file` with `_as_binary_file` so I wonder if it's worth calling `_as_binary_file` inside `_as_upload_file` just to clean this up a bit and make the caller's code a bit simpler?